tests/test_endpoints.py

```diff
@@ -1937,3 +2114,20 @@ async def test_with_7_x_firmware(
             await envoy.disable_charge_from_grid()
 
     assert envoy.phase_count == phase_count
+    assert envoy._common_properties.ct_meter_count == common_properties["ctMeters"]
+    assert envoy._common_properties.phase_count == common_properties["phaseCount"]
+    assert envoy._common_properties.phase_mode == common_properties["phaseMode"]
+    assert (
+        envoy._common_properties.consumption_meter_type
+        == common_properties["consumptionMeter"]
+    )
+
+    if production_phases is None:
+        assert data.system_production_phases is None
+    if consumption_phases is None:
+        assert data.system_consumption_phases is None
+
+    if data.system_production_phases is None:
+        assert not production_phases
+    if envoy.data.system_consumption_phases is None:
+        assert not consumption_phases
```

Comment:
We usually don't want branching in tests as it makes them harder to maintain

Reply:
I see. But I need to deal with some condtional situations, much like tests considering supported_features. Tried to come up with some logic for the assert to avoid the if but it would make the assert condition way more complex as the if statement. For now I can remove them as there's no data yet and I'll give it another shot with the data populated.

Reply:
Few options here.

1. Remove the test (probably best choice for now)
2. Use pytest.mark.parametrize
3. Split the test into separate tests

Reply:
For now I've removed them.